physics.py

```diff
@@ -171,12 +178,96 @@ def update(self, tm_diff: float, robotVoltage: float) -> None:
             [chassisSpeed.vx, chassisSpeed.vy, chassisSpeed.omega],
         )
 
-        deltaTrans = Transform2d(deltaX, deltaY, deltaHeading)
+        deltaTrans = Transform2d(deltaX, -deltaY, deltaHeading)
 
         newPose = self.pose + deltaTrans
         self.pose = newPose
 
 
+class NoteSim:
+    def __init__(self) -> None:
+        self.midlineNotes = constants.kNotesStartingMidline
+        self.blueNotes = constants.kNotesStartingBlueWing
+        self.redNotes = constants.kNotesStartingRedWing
+
+        self.loadingNotes = [
+            constants.kNoteLoadingStationPositionBlue,
+            constants.kNoteLoadingStationPositionRed,
+        ]
+
+    def canPickup(self, note: Pose3d, botPose) -> bool:
+        if pointInCircle(botPose.translation(), note.toPose2d().translation(), 0.5):
+            return True
+        return False
+
+    def update(self, _tm_diff, bot: MentorBot):
+        SmartDashboard.putNumberArray(
+            constants.kSimNotePositionsKey,
+            convertToSendablePoses(
+                [
+                    *self.midlineNotes,
+                    *self.blueNotes,
+                    *self.redNotes,
+                    *self.loadingNotes,
+                ]
+            ),
+        )
+
+        # check whether intaking, update sensors according to position on field
+
+        intaking = bot.container.intake.state == IntakeSubsystem.IntakeState.Intaking
+
+        botPose = Pose2d(
+            *SmartDashboard.getNumberArray(constants.kSimRobotPoseArrayKey, [0, 0, 0])
+        )
+
+        hasNote = SmartDashboard.getBoolean(
```

Comment:
Why not take this from intake subsystem itself (have intakesub post to smartdash and then take) instead of directly from a sensor?